### ISSUE TITLE
Remove dependency on docker CLI

### ISSUE THREAD

**Issue by shizhMSFT** [oras-project/oras#414]
`oras` CLI leverage `github.com/docker/cli` for reading / writing docker config file for login and logout. It is possible that using an old version of oras CLI with a new version of docker CLI at the same time may cause loss on some config fields.

**Comment by FeynmanZhou** [comment]
@shizhMSFT Does it mean ORAS CLI will need to implement its own authentication system?

**Comment by shizhMSFT** [comment]
> @shizhMSFT Does it mean ORAS CLI will need to implement its own authentication system?

No, the new implementation will still be compatible with the `config.json` of the docker CLI.

**Comment by TerryHowe** [comment]
There is an indirect dependency I think

**Comment by qweeah** [comment]
ORAS CLI still have deps on github.com/docker/docker-credential-helpers v0.7.0. Resolving this issue is depending on https://github.com/oras-project/oras-credentials-go/issues/67 and merging of https://github.com/oras-project/oras-credentials-go/pull/68
